plugins/ipam/ipstore/ipstore.go

```diff
@@ -159,6 +167,53 @@ func (manager *IPManager) Release(ip string) error {
 	return nil
 }
 
+// ReleaseByValue release the key-value pair by value
+func (manager *IPManager) ReleaseByValue(value string) (string, error) {
+	// TODO improve this part by implement listing all the kv pairs
+	// libkv library only provide list all key-value pairs with prefix of key, and
+	// if the result is empty it will return store.ErrKeyNotFound error
+	kvPairs, err := manager.client.List("1")
+	if err == store.ErrKeyNotFound {
+		return "", errors.Errorf("releaseByValue ipstore: no ip associated with the id: %s", value)
+	}
+	if err != nil {
+		return "", errors.Wrapf(err, "releaseByValue ipstore: failed to list the key-value pairs in db")
+	}
+
+	var ipDeleted string
+	for _, kvPair := range kvPairs {
+		if string(kvPair.Value) == value {
+			err = manager.Release(kvPair.Key)
+			if err != nil {
+				return "", err
+			}
+			ipDeleted = kvPair.Key
+			return ipDeleted, nil
+		}
+	}
+	return "", errors.Errorf("No ip associated with the given value: %s", value)
```

Comment:
Can you reformat the error message as `"release ipstore: no ip address associated ..."`